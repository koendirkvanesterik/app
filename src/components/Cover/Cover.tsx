import classNames from 'classnames'
import React from 'react'

import { Scene } from './components/Scene'

export const Cover: React.FunctionComponent = ({ children }) => (
  <div
    className={classNames([
      'border',
      'border-warmGray-800',
      'h-96',
      'relative',
      'rounded-lg'
    ])}
  >
    <div className={classNames(['absolute', 'h-full', 'w-full'])}>
      <Scene />
    </div>
    <div
      className={classNames([
        'flex',
        'h-full',
        'items-end',
        'p-10',
        'relative',
        'z-10',
      ])}
    >
      <div className={classNames(['w-4/12'])}>{children}</div>
    </div>
  </div>
)
