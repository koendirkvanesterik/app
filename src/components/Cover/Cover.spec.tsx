import { render } from '@testing-library/react'
import React from 'react'
import { Cover } from './Cover'

jest.mock('./components', () => ({
  Scene: () => <div />,
}))

it('should render as expected', () => {
  const { container } = render(<Cover />)
  expect(container).toMatchSnapshot()
})

afterAll(() => {
  jest.resetAllMocks()
})
