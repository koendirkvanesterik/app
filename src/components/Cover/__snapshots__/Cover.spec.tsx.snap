// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="border border-stone-800 h-96 relative rounded-lg"
  >
    <div
      class="absolute h-full w-full"
    >
      <div />
    </div>
    <div
      class="flex h-full items-end relative z-10"
    />
  </div>
</div>
`;
